Tests/StoreKitUnitTests/ProductsManagerTests.swift

```diff
@@ -126,3 +128,75 @@ class ProductsManagerTests: StoreKitConfigTestCase {
     }
 
 }
+
+@available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+class ProductsManagerDiagnosticsTrackingTests: ProductsManagerTests {
+
+    private var mockDiagnosticsTracker: MockDiagnosticsTracker!
+
+    private var productsManager: ProductsManager!
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+
+        self.mockDiagnosticsTracker = MockDiagnosticsTracker()
+    }
+
+    func testFetchProductsWithIdentifiersSK1TracksCorrectly() throws {
+        let manager = self.createManager(storeKitVersion: .storeKit1,
+                                         diagnosticsTracker: self.mockDiagnosticsTracker)
+
+        let identifier = "com.revenuecat.monthly_4.99.1_week_intro"
+        _ = waitUntilValue(timeout: Self.requestDispatchTimeout) { completed in
+            manager.products(withIdentifiers: Set([identifier]), completion: completed)
+        }
+
+        expect(self.mockDiagnosticsTracker.trackedProductsRequestParams.count) == 1
+        let params = self.mockDiagnosticsTracker.trackedProductsRequestParams.first
+        expect(params?.wasSuccessful) == true
+        expect(params?.storeKitVersion) == .storeKit1
+        expect(params?.errorMessage) == nil
+        expect(params?.errorCode) == nil
+    }
+
+    func testFetchProductsWithIdentifiersSK2TracksCorrectly() throws {
+        let manager = self.createManager(storeKitVersion: .storeKit2,
+                                         diagnosticsTracker: self.mockDiagnosticsTracker)
+
+        let identifier = "com.revenuecat.monthly_4.99.1_week_intro"
+        _ = waitUntilValue(timeout: Self.requestDispatchTimeout) { completed in
+            manager.products(withIdentifiers: Set([identifier]), completion: completed)
+        }
+
+        expect(self.mockDiagnosticsTracker.trackedProductsRequestParams.count) == 1
+        let params = self.mockDiagnosticsTracker.trackedProductsRequestParams.first
+        expect(params?.wasSuccessful) == true
+        expect(params?.storeKitVersion) == .storeKit2
+        expect(params?.errorMessage) == nil
+        expect(params?.errorCode) == nil
```

Comment:
I think these should be `to(beNil))`

Reply:
Good catch! Changed in https://github.com/RevenueCat/purchases-ios/pull/4247/commits/e81cd1c79681b2229f9d6abc2caad44329368982